code/modules/admin/job_manager.dm

```diff
@@ -0,0 +1,78 @@
+
+/datum/job_manager
+
+/datum/job_manager/ui_state(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/job_manager/ui_status(mob/user)
+  return tgui_admin_state.can_use_topic(src, user)
+
+/datum/job_manager/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "JobManager")
+	ui.open()
+
+/datum/job_manager/ui_data(mob/user)
+	var/list/staple_job_data = list()
+	var/list/special_job_data = list()
+	var/list/hidden_job_data = list()
+	for (var/datum/job/JOB in job_controls.staple_jobs)
+		staple_job_data += list(list(name = JOB.name, type = JOB.job_category, count = countJob(JOB.name), limit = JOB.limit))
+	for (var/datum/job/JOB in job_controls.special_jobs)
+		special_job_data += list(list(name = JOB.name, type = JOB.job_category, count = countJob(JOB.name), limit = JOB.limit))
+	for (var/datum/job/JOB in job_controls.hidden_jobs)
+		hidden_job_data += list(list(name = JOB.name, type = JOB.job_category, count = countJob(JOB.name), limit = JOB.limit))
+	. = list(
+			"stapleJobs" = staple_job_data,
+			"specialJobs" = special_job_data,
+			"hiddenJobs" = hidden_job_data,
+			"allowSpecialJobs" = job_controls.allow_special_jobs
+		)
+
+/datum/job_manager/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	if (!isadmin(ui.user))
+		return
+	switch(action)
+		if ("alter_cap")
+			var/datum/job/JOB = find_job_in_controller_by_string(params["job"])
```

Comment:
Prefer non-all-caps names for local variables, all caps implies a global constant.